### PR TITLE
[sandbox][evomorphey-work] Adding mongodb.

### LINKED ISSUE #4: [FEATURE]: Create and add MongoDB.
**Description**
---
Need to create MongoDB database. Create collection for one game. Create classes for work with MongoDB in C#.

**Expected Result**
---
- MongoDB is created
- We can fetch all info from MongoDB

### PR DESCRIPTION
**Pull request by 3voMorphey**

### CODE REVIEW

File: Domain.CardGamesGuruMiniApp/Entities/GameEntities/Game.cs [hunk 1]
```diff
@@ -1,7 +1,18 @@
...
     public Guid Id { get; set; }
+    public string NameIndex { get; set; }
     public string Name { get; set; }
+    public string Description { get; set; }
+    public DateTime CreatedDate { get; set; }
+    public DateTime UpdatedDate { get; set; }
+    public GameType GameType { get; set; }
+
+
+    // image? and other fields?
```
Comment (Alkamal30) [inline Domain.CardGamesGuruMiniApp/Entities/GameEntities/Game.cs:16]: What is it?

File: Domain.CardGamesGuruMiniApp/Entities/GameEntities/Game.cs [hunk 2]
```diff
@@ -1,7 +1,18 @@
+using Domain.CardGamesGuruMiniApp.Enums.GameEnums;
+
 namespace Domain.CardGamesGuruMiniApp.Entities.Game.GameEntities;
 
 public class Game
 {
     public Guid Id { get; set; }
+    public string NameIndex { get; set; }
```
Comment (Alkamal30) [inline Domain.CardGamesGuruMiniApp/Entities/GameEntities/Game.cs:8]: What does it mean this property?

File: Domain.CardGamesGuruMiniApp/Configuration/MongoDbOptions.cs [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Domain.CardGamesGuruMiniApp.Configuration
+{
+    public class MongoDbOptions
+    {
+        public string ConnectionString { get; set; }
+        public string DataBaseName { get; set; }
```
Comment (Alkamal30) [inline Domain.CardGamesGuruMiniApp/Configuration/MongoDbOptions.cs]: May be `DatabaseName`?

File: Infrastructure.CardGamesGuruMiniApp/Repositories/BaseRepository.cs [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Infrastructure.CardGamesGuruMiniApp.Repositories
+{
+    public abstract class BaseRepository<T>
+    {
+        protected readonly IMongoDatabase database;
+        //private readonly string collection;
+        private readonly MongoCollectionSettings mongoCollectionSettings;
+
+        protected BaseRepository(IMongoDatabase database, string collection, IOptions<MongoDbOptions> mongoDbOptions)
```
Comment (Alkamal30) [inline Infrastructure.CardGamesGuruMiniApp/Repositories/BaseRepository.cs]: `collection` is not used

File: Infrastructure.CardGamesGuruMiniApp/Models/GamesModels/Game.cs
Comment (Alkamal30) [inline Infrastructure.CardGamesGuruMiniApp/Models/GamesModels/Game.cs:1]: File name and class name are not equals

File: Infrastructure.CardGamesGuruMiniApp/Repositories/BaseRepository.cs [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+        }
+
+        protected BaseRepository(IMongoDatabase database,IOptions<MongoDbOptions> mongoDbOptions)
+        {
+            this.database = database;
+            var wcTimeout = TimeSpan.FromMilliseconds(mongoDbOptions.Value.Timeout);
+            this.mongoCollectionSettings = new() { WriteConcern = new WriteConcern("majority", wcTimeout, null, true) };
+        }
+
+        protected IMongoCollection<T> Items => database.GetCollection<T>("games", mongoCollectionSettings);
```
Comment (Alkamal30) [inline Infrastructure.CardGamesGuruMiniApp/Repositories/BaseRepository.cs]: Constant `games` in Base repository?

File: Service.CardGamesGuruMiniApp/Handlers/GameHandler/CreateGameHandler.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
+﻿using Domain.CardGamesGuruMiniApp.Entities.Game.GameEntities;
+using Domain.CardGamesGuruMiniApp.Enums.GameEnums;
+using MediatR;
+using Services.CardGamesGuruMiniApp.Services.GameService.Interfaces;
+using System.Text.Json.Serialization;
+
+namespace Services.CardGamesGuruMiniApp.Handlers.GameHandler
+{
+
+    public class CreateGameQuery : IRequest
```
Comment (Alkamal30) [inline Service.CardGamesGuruMiniApp/Handlers/GameHandler/CreateGameHandler.cs:10]: Move to separate file?

File: Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs [hunk 7]
```diff
@@ -1,10 +1,40 @@
...
+        var item = new GameBson()
+        {
+            Name = game.Name,
+            Description = game.Description,
+            NameIndex = game.NameIndex,
+            CreatedDate = game.CreatedDate,
+        };
+        item.Id = game.Id.ToString();
+
+        //item = mapper.Map<GameBson>(game);
```
Comment (Alkamal30) [inline Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs:31]: Remove comment

File: Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs [hunk 8]
```diff
@@ -1,10 +1,40 @@
...
 
 namespace Services.CardGamesGuruMiniApp.Services.GameService;
 
 public class GameService : IGameService
 {
+    private readonly IGameRepository gameRepository;
+    private readonly IMapper mapper;
+    public GameService(IGameRepository gameRepository, IMapper mapper)
+    {
+        this.gameRepository = gameRepository;
```
Comment (Alkamal30) [inline Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs:16]: Don't use `this` pls

File: Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs [hunk 9]
```diff
@@ -1,10 +1,40 @@
...
+using Infrastructure.CardGamesGuruMiniApp.Repositories.Interfaces;
+using Infrastructure.CardGamesGuruMiniApp.Models.GamesModels;
+using MongoDB.Bson;
+using AutoMapper;
 
 namespace Services.CardGamesGuruMiniApp.Services.GameService;
 
 public class GameService : IGameService
 {
+    private readonly IGameRepository gameRepository;
```
Comment (Alkamal30) [inline Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs:12]: Bad naming. Follow [microsoft naming conventions](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#naming-conventions) pls

File: Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs (hunk 8 above)
Comment (3voMorphey) [reply Service.CardGamesGuruMiniApp/Services/GameService/GameService.cs:16]: пошел нахуй

**Review by Alkamal30 — APPROVED** [review]

**Pull request merged**